tests/components/dhcp/test_init.py

```diff
@@ -659,3 +661,48 @@ async def test_aiodiscover_does_not_call_again_on_shorter_hostname(hass):
         dhcp.HOSTNAME: "irobot-abcdef",
         dhcp.MAC_ADDRESS: "b8b7f16db533",
     }
+
+
+async def test_aiodiscover_finds_new_hosts_after_interval(hass):
+    """Test aiodiscover finds new host after interval."""
+    with patch.object(
+        hass.config_entries.flow, "async_init", return_value=mock_coro()
```

Comment:
Doesn't patch.object patch a coroutine function by default with an AsyncMock? If not we should use `AsyncMock` instead of `mock_coro` maybe as a side effect. `mock_coro` is legacy.

Reply:
Absolutely it does now. I cleaned them all up and rebased to fix the conflict.

Reply:
I fixed https://github.com/home-assistant/core/pull/48242/files as well